### PR TITLE
Prettier, commit lint, commitizen, husky, tslint upgrade

### PR DESCRIPTION
**Pull request by ike18t**

### CODE REVIEW

File: lib/common/jit-reflector.ts [hunk 1]
```diff
@@ -12,14 +12,10 @@ export class JitReflector implements CompileReflector {
     return this._reflectionCapabilities.annotations(typeOrFunc);
   }
 
-  componentModuleUrl(type: any, cmpMetadata: Component): string { /* tslint:disable-line prefer-function-over-method */
-    return `./${stringify(type)}`;
-  }
+  componentModuleUrl = (type: unknown) => `./${stringify(type)}`;
```
Comment (satanTime) [inline lib/common/jit-reflector.ts]: as far as I remember `unknown` doesn't work on older TS versions. But it is awesome!

**Comment by satanTime** [comment]
I would vote for trailing comma.

File: lib/common/jit-reflector.ts (hunk 1 above)
Comment (ike18t) [reply lib/common/jit-reflector.ts]: Good catch!

**Comment by ike18t** [comment]
> I would vote for trailing comma.

I hate "dangles" but you and @getsaf would have me outvoted 2 to 1.

**Comment by satanTime** [comment]
I like less diff when someone adds one more item somewhere.

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "bracketSpacing": true,
+  "semi": true,
+  "printWidth": 120,
+  "singleQuote": true,
+  "arrowParens": "avoid",
+  "trailingComma": "es5"
```
Comment (ike18t) [inline .prettierrc.json]: @satanTime @getsaf because I'm a team player.

**Pull request merged**